conans/client/remote_manager.py

```diff
@@ -150,24 +157,29 @@ def get_package(self, pref, dest_folder, remote, output, recorder):
 
             duration = time.time() - t1
             log_package_download(pref, duration, remote, zipped_files)
-            unzip_and_get_files(zipped_files, dest_folder, PACKAGE_TGZ_NAME, output=self._output)
+            unzip_and_get_files(zipped_files, unzip_folder, PACKAGE_TGZ_NAME,
+                                output=self._output,
+                                remove_tgz=not bool(package_install_folder))
             # Issue #214 https://github.com/conan-io/conan/issues/214
-            touch_folder(dest_folder)
+            touch_folder(unzip_folder)
             if get_env("CONAN_READ_ONLY_CACHE", False):
-                make_read_only(dest_folder)
+                make_read_only(package_folder)
             recorder.package_downloaded(pref, remote.url)
             output.success('Package installed %s' % pref.id)
         except NotFoundException:
             raise PackageNotFoundException(pref)
         except BaseException as e:
             output.error("Exception while getting package: %s" % str(pref.id))
             output.error("Exception: %s %s" % (type(e), str(e)))
-            try:
-                output.warn("Trying to remove package folder: %s" % dest_folder)
-                rmdir(dest_folder)
-            except OSError as e:
-                raise ConanException("%s\n\nCouldn't remove folder '%s', might be busy or open. "
-                                     "Close any app using it, and retry" % (str(e), dest_folder))
+            for folder in package_folder, unzip_folder:
```

Comment:
Maybe... and avoid the `if package_folder == unzip_folder:` below

```suggestion
            for folder in set(package_folder, unzip_folder):
```